[Bug]: types are no longer compatible with `"module": "Node16"`

### Reproduction link or steps

1. Open https://stackblitz.com/edit/github-nci4o6nk-oxwjowuk?file=foo.ts,tsconfig.json,package.json
2. `npm i` runs automatically
3. `npm run check` runs automatically
4. Error happens

### What is expected?

No error happens.

### What is actually happening?

The following error happens:
```
node_modules/rolldown/dist/types/index.d.ts:1:84 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './types/rolldown-output.js'?

1 import { RolldownOutput, OutputAsset, OutputChunk, RenderedChunk, SourceMap } from './types/rolldown-output';
                                                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~
```

<details>
<summary>Full output</summary>


```
node_modules/rolldown/dist/types/index.d.ts:1:84 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './types/rolldown-output.js'?

1 import { RolldownOutput, OutputAsset, OutputChunk, RenderedChunk, SourceMap } from './types/rolldown-output';
                                                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:2:76 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './options/input-options.js'?

2 import type { InputOptions, InputOption, ExternalOption, JsxOptions } from './options/input-options';
                                                                             ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:3:50 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './options/output-options.js'?

3 import type { ModuleFormat, OutputOptions } from './options/output-options';
                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:4:38 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './types/rolldown-options.js'?

4 import type { RolldownOptions } from './types/rolldown-options';
                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:5:295 - error TS2834: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

5 import type { AsyncPluginHooks, CustomPluginOptions, FunctionPluginHooks, ImportKind, LoadResult, ModuleOptions, ModuleType, ObjectHook, ParallelPluginHooks, PartialResolvedId, Plugin, RolldownPlugin, RolldownPluginOption, ResolveIdResult, ResolvedId, SourceDescription, TransformResult } from './plugin';
                                                                                                                                                                                                                                                                                                        ~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:6:44 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './plugin/parallel-plugin.js'?

6 import { DefineParallelPluginResult } from './plugin/parallel-plugin';
                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:7:30 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './utils/define-config.js'?

7 import { defineConfig } from './utils/define-config';
                               ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:8:26 - error TS2834: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

8 import { rolldown } from './api/rolldown';
                           ~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:9:23 - error TS2834: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

9 import { watch } from './api/watch';
                        ~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:10:30 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './types/config-export.js'?

10 import { ConfigExport } from './types/config-export';
                                ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:11:31 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './api/rolldown/rolldown-build.js'?

11 import { RolldownBuild } from './api/rolldown/rolldown-build';
                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:12:58 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './plugin/plugin-context.js'?

12 import { EmittedAsset, EmittedFile, PluginContext } from './plugin/plugin-context';
                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:13:40 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './plugin/transform-plugin-context.js'?

13 import { TransformPluginContext } from './plugin/transform-plugin-context';
                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:14:63 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './options/normalized-output-options.js'?

14 import { InternalModuleFormat, NormalizedOutputOptions } from './options/normalized-output-options';
                                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:15:34 - error TS2834: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

15 import { PreRenderedChunk } from './binding';
                                    ~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:16:29 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './types/utils.js'?

16 import { PartialNull } from './types/utils';
                               ~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:17:40 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './options/normalized-input-options.js'?

17 import { NormalizedInputOptions } from './options/normalized-input-options';
                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:18:28 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './types/module-info.js'?

18 import { ModuleInfo } from './types/module-info';
                              ~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:19:38 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './plugin/minimal-plugin-context.js'?

19 import { MinimalPluginContext } from './plugin/minimal-plugin-context';
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:20:54 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './types/sourcemap.js'?

20 import { ExistingRawSourceMap, SourceMapInput } from './types/sourcemap';
                                                        ~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:21:30 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './types/output-bundle.js'?

21 import { OutputBundle } from './types/output-bundle';
                                ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:22:30 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './options/watch-options.js'?

22 import { WatchOptions } from './options/watch-options';
                                ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:23:33 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './api/watch/watch-emitter.js'?

23 import { RolldownWatcher } from './api/watch/watch-emitter';
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:24:42 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './api/build.js'?

24 import { build, type BuildOptions } from './api/build';
                                            ~~~~~~~~~~~~~

node_modules/rolldown/dist/types/index.d.ts:28:62 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './types/misc.js'?

28 export type { RollupError, RollupLog, LoggingFunction } from './types/misc';
                                                                ~~~~~~~~~~~~~~


Found 25 errors in the same file, starting at: node_modules/rolldown/dist/types/index.d.ts:1
```


</details>


### System Info

```Shell
Stackblitz

  System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    rolldown: 1.0.0-beta.2-commit.5deb279 => 1.0.0-beta.2-commit.5deb279
```

### Any additional comments?

No error happens with 1.0.0-beta.1.

I found this when upgrading rolldown in rolldown-vite. Vite has a type check script to ensure the types are compatible with stricter options like `"module": "node16"`.

--- a/cspell.json
+++ b/cspell.json
@@ -21,6 +21,7 @@
     "/packages/rolldown/npm",
     "scripts/snap-diff/**/*.{txt,md}",
     "packages/rolldown/src/binding.cjs",
+    "packages/rolldown/src/binding.js",
     "/CHANGELOG.md",
     "/crates/rolldown/tests/esbuild/splitting/ignore-test.md",
     "/packages/rollup-tests/src/ignored-by-unsupported-features.md",

--- a/packages/rolldown/build.ts
+++ b/packages/rolldown/build.ts
@@ -150,7 +150,7 @@ const configs = defineConfig([
       },
 
       {
-        name: 'cleanup binding.cjs',
+        name: 'cleanup binding.js',
         transform: {
           filter: {
             code: {
@@ -158,7 +158,7 @@ const configs = defineConfig([
             },
           },
           handler(code, id) {
-            if (id.endsWith('binding.cjs')) {
+            if (id.endsWith('binding.js')) {
               const ret = code.replace(
                 'require = createRequire(__filename)',
                 '',
@@ -198,6 +198,8 @@ const configs = defineConfig([
   },
 ])
 
-for (const config of configs) {
-  await (await rolldown(config)).write(config.output as OutputOptions)
-}
+;(async () => {
+  for (const config of configs) {
+    await (await rolldown(config)).write(config.output as OutputOptions)
+  }
+})()

--- a/packages/rolldown/package.json
+++ b/packages/rolldown/package.json
@@ -26,7 +26,6 @@
   "bin": {
     "rolldown": "./bin/cli.js"
   },
-  "type": "module",
   "main": "./dist/cjs/index.cjs",
   "types": "./dist/types/index.d.ts",
   "exports": {
@@ -62,19 +61,19 @@
   "scripts": {
     "# Scrips for binding #": "_",
     "artifacts": "napi artifacts --cwd ./src --package-json-path ../package.json -o=../artifacts --npm-dir ../npm",
-    "build-binding": "napi build -o=./src --manifest-path ../../crates/rolldown_binding/Cargo.toml --platform -p rolldown_binding --js binding.cjs --dts binding.d.ts --no-const-enum",
+    "build-binding": "napi build -o=./src --manifest-path ../../crates/rolldown_binding/Cargo.toml --platform -p rolldown_binding --js binding.js --dts binding.d.ts --no-const-enum",
     "build-binding:release": "pnpm build-binding --release",
     "build-binding:wasi": "pnpm build-binding --target wasm32-wasip1-threads",
     "build-binding:wasi:release": "pnpm build-binding --profile release-wasi --target wasm32-wasip1-threads",
     "# Scrips for node #": "_",
     "bak_build-node": "unbuild",
-    "build-node": "node --import @oxc-node/core/register ./build.ts",
+    "build-node": "tsx ./build.ts",
     "build-types": "tsc -p ./tsconfig.json",
-    "build-native:debug": "run-s build-binding build-types build-node build-type-check",
-    "build-native:release": "run-s build-binding:release build-types build-node build-type-check",
+    "build-native:debug": "run-s build-binding build-types build-node build-types-check",
+    "build-native:release": "run-s build-binding:release build-types build-node build-types-check",
     "build-wasi:debug": "run-s build-binding build-binding:wasi build-node",
     "build-wasi:release": "run-s build-binding build-binding:wasi:release build-node",
-    "build-type-check": "tsc -p ./tsconfig.check.json",
+    "build-types-check": "tsc -p ./tsconfig.check.json",
     "# Scrips for docs #": "_",
     "extract-options-doc": "typedoc"
   },

--- a/packages/rolldown/package.json
+++ b/packages/rolldown/package.json
@@ -70,10 +70,11 @@
     "bak_build-node": "unbuild",
     "build-node": "node --import @oxc-node/core/register ./build.ts",
     "build-types": "tsc -p ./tsconfig.json",
-    "build-native:debug": "run-s build-binding build-types build-node",
-    "build-native:release": "run-s build-binding:release build-types build-node",
+    "build-native:debug": "run-s build-binding build-types build-node build-type-check",
+    "build-native:release": "run-s build-binding:release build-types build-node build-type-check",
     "build-wasi:debug": "run-s build-binding build-binding:wasi build-node",
     "build-wasi:release": "run-s build-binding build-binding:wasi:release build-node",
+    "build-type-check": "tsc -p ./tsconfig.check.json",
     "# Scrips for docs #": "_",
     "extract-options-doc": "typedoc"
   },

--- a/packages/rolldown/src/binding.d.ts
+++ b/packages/rolldown/src/binding.d.ts
@@ -745,6 +745,12 @@ export interface EcmaScriptModule {
   importMetas: Array<Span>
 }
 
+export interface ErrorLabel {
+  message?: string
+  start: number
+  end: number
+}
+
 export interface Es2015Options {
   /** Transform arrow functions into function expressions. */
   arrowFunction?: ArrowFunctionsOptions
@@ -1007,6 +1013,13 @@ export interface OverwriteOptions {
   contentOnly: boolean
 }
 
+export interface OxcError {
+  severity: Severity
+  message: string
+  labels: Array<ErrorLabel>
+  helpMessage?: string
+}
+
 /**
  * Parse asynchronously.
  *
@@ -1066,6 +1079,21 @@ export interface ReactRefreshOptions {
 }
 
 export declare function registerPlugins(id: number, plugins: Array<BindingPluginWithIndex>): void
+
+export type Severity =  'Error'|
+'Warning'|
+'Advice';
+
+export interface SourceMap {
+  file?: string
+  mappings: string
+  names: Array<string>
+  sourceRoot?: string
+  sources: Array<string>
+  sourcesContent?: Array<string>
+  version: number
+  x_google_ignoreList?: Array<number>
+}
 
 export interface SourceMapOptions {
   includeContent?: boolean

--- a/packages/rolldown/src/binding.js
+++ b/packages/rolldown/src/binding.js
@@ -398,4 +398,5 @@ module.exports.parseAsync = nativeBinding.parseAsync
 module.exports.parseSync = nativeBinding.parseSync
 module.exports.parseWithoutReturn = nativeBinding.parseWithoutReturn
 module.exports.registerPlugins = nativeBinding.registerPlugins
+module.exports.Severity = nativeBinding.Severity
 module.exports.transform = nativeBinding.transform

--- a/packages/rolldown/src/rolldown-binding.wasi-browser.js
+++ b/packages/rolldown/src/rolldown-binding.wasi-browser.js
@@ -58,28 +58,32 @@ const {
 })
 
 function __napi_rs_initialize_modules(__napiInstance) {
+  __napiInstance.exports['__napi_register__OxcError_struct_0']?.()
+  __napiInstance.exports['__napi_register__ErrorLabel_struct_1']?.()
+  __napiInstance.exports['__napi_register__Severity_2']?.()
+  __napiInstance.exports['__napi_register__SourceMap_struct_0']?.()
   __napiInstance.exports['__napi_register__IsolatedDeclarationsResult_struct_0']?.()
+  __napiInstance.exports['__napi_register__MagicString_struct_0']?.()
   __napiInstance.exports['__napi_register__IsolatedDeclarationsOptions_struct_1']?.()
   __napiInstance.exports['__napi_register__isolated_declaration_2']?.()
-  __napiInstance.exports['__napi_register__TransformResult_struct_3']?.()
-  __napiInstance.exports['__napi_register__MagicString_struct_0']?.()
-  __napiInstance.exports['__napi_register__TransformOptions_struct_4']?.()
-  __napiInstance.exports['__napi_register__CompilerAssumptions_struct_5']?.()
-  __napiInstance.exports['__napi_register__TypeScriptOptions_struct_6']?.()
   __napiInstance.exports['__napi_register__LineColumn_struct_1']?.()
   __napiInstance.exports['__napi_register__OverwriteOptions_struct_2']?.()
   __napiInstance.exports['__napi_register__SourceMapOptions_struct_3']?.()
+  __napiInstance.exports['__napi_register__TransformResult_struct_3']?.()
   __napiInstance.exports['__napi_register__GenerateDecodedMapOptions_struct_4']?.()
+  __napiInstance.exports['__napi_register__TransformOptions_struct_4']?.()
+  __napiInstance.exports['__napi_register__CompilerAssumptions_struct_5']?.()
+  __napiInstance.exports['__napi_register__TypeScriptOptions_struct_6']?.()
   __napiInstance.exports['__napi_register__JsxOptions_struct_7']?.()
   __napiInstance.exports['__napi_register__ReactRefreshOptions_struct_8']?.()
+  __napiInstance.exports['__napi_register__MagicString_impl_24']?.()
   __napiInstance.exports['__napi_register__ArrowFunctionsOptions_struct_9']?.()
   __napiInstance.exports['__napi_register__Es2015Options_struct_10']?.()
   __napiInstance.exports['__napi_register__Helpers_struct_11']?.()
   __napiInstance.exports['__napi_register__HelperMode_12']?.()
-  __napiInstance.exports['__napi_register__transform_13']?.()
-  __napiInstance.exports['__napi_register__MagicString_impl_24']?.()
   __napiInstance.exports['__napi_register__ParserOptions_struct_25']?.()
   __napiInstance.exports['__napi_register__ParseResult_struct_26']?.()
+  __napiInstance.exports['__napi_register__transform_13']?.()
   __napiInstance.exports['__napi_register__ParseResult_impl_32']?.()
   __napiInstance.exports['__napi_register__Comment_struct_33']?.()
   __napiInstance.exports['__napi_register__EcmaScriptModule_struct_34']?.()
@@ -236,4 +240,5 @@ export const parseAsync = __napiModule.exports.parseAsync
 export const parseSync = __napiModule.exports.parseSync
 export const parseWithoutReturn = __napiModule.exports.parseWithoutReturn
 export const registerPlugins = __napiModule.exports.registerPlugins
+export const Severity = __napiModule.exports.Severity
 export const transform = __napiModule.exports.transform

--- a/packages/rolldown/src/rolldown-binding.wasi.cjs
+++ b/packages/rolldown/src/rolldown-binding.wasi.cjs
@@ -82,28 +82,32 @@ const { instance: __napiInstance, module: __wasiModule, napiModule: __napiModule
 })
 
 function __napi_rs_initialize_modules(__napiInstance) {
+  __napiInstance.exports['__napi_register__OxcError_struct_0']?.()
+  __napiInstance.exports['__napi_register__ErrorLabel_struct_1']?.()
+  __napiInstance.exports['__napi_register__Severity_2']?.()
+  __napiInstance.exports['__napi_register__SourceMap_struct_0']?.()
   __napiInstance.exports['__napi_register__IsolatedDeclarationsResult_struct_0']?.()
+  __napiInstance.exports['__napi_register__MagicString_struct_0']?.()
   __napiInstance.exports['__napi_register__IsolatedDeclarationsOptions_struct_1']?.()
   __napiInstance.exports['__napi_register__isolated_declaration_2']?.()
-  __napiInstance.exports['__napi_register__TransformResult_struct_3']?.()
-  __napiInstance.exports['__napi_register__MagicString_struct_0']?.()
-  __napiInstance.exports['__napi_register__TransformOptions_struct_4']?.()
-  __napiInstance.exports['__napi_register__CompilerAssumptions_struct_5']?.()
-  __napiInstance.exports['__napi_register__TypeScriptOptions_struct_6']?.()
   __napiInstance.exports['__napi_register__LineColumn_struct_1']?.()
   __napiInstance.exports['__napi_register__OverwriteOptions_struct_2']?.()
   __napiInstance.exports['__napi_register__SourceMapOptions_struct_3']?.()
+  __napiInstance.exports['__napi_register__TransformResult_struct_3']?.()
   __napiInstance.exports['__napi_register__GenerateDecodedMapOptions_struct_4']?.()
+  __napiInstance.exports['__napi_register__TransformOptions_struct_4']?.()
+  __napiInstance.exports['__napi_register__CompilerAssumptions_struct_5']?.()
+  __napiInstance.exports['__napi_register__TypeScriptOptions_struct_6']?.()
   __napiInstance.exports['__napi_register__JsxOptions_struct_7']?.()
   __napiInstance.exports['__napi_register__ReactRefreshOptions_struct_8']?.()
+  __napiInstance.exports['__napi_register__MagicString_impl_24']?.()
   __napiInstance.exports['__napi_register__ArrowFunctionsOptions_struct_9']?.()
   __napiInstance.exports['__napi_register__Es2015Options_struct_10']?.()
   __napiInstance.exports['__napi_register__Helpers_struct_11']?.()
   __napiInstance.exports['__napi_register__HelperMode_12']?.()
-  __napiInstance.exports['__napi_register__transform_13']?.()
-  __napiInstance.exports['__napi_register__MagicString_impl_24']?.()
   __napiInstance.exports['__napi_register__ParserOptions_struct_25']?.()
   __napiInstance.exports['__napi_register__ParseResult_struct_26']?.()
+  __napiInstance.exports['__napi_register__transform_13']?.()
   __napiInstance.exports['__napi_register__ParseResult_impl_32']?.()
   __napiInstance.exports['__napi_register__Comment_struct_33']?.()
   __napiInstance.exports['__napi_register__EcmaScriptModule_struct_34']?.()
@@ -260,4 +264,5 @@ module.exports.parseAsync = __napiModule.exports.parseAsync
 module.exports.parseSync = __napiModule.exports.parseSync
 module.exports.parseWithoutReturn = __napiModule.exports.parseWithoutReturn
 module.exports.registerPlugins = __napiModule.exports.registerPlugins
+module.exports.Severity = __napiModule.exports.Severity
 module.exports.transform = __napiModule.exports.transform

--- a/packages/rolldown/tsconfig.check.json
+++ b/packages/rolldown/tsconfig.check.json
@@ -12,6 +12,5 @@
     "exactOptionalPropertyTypes": true,
     "resolveJsonModule": true
   },
-  "include": ["dist/types/**/*.d.ts"],
-  "exclude": []
+  "include": ["dist/types/**/*.d.ts"]
 }

--- a/packages/rolldown/tsconfig.check.json
+++ b/packages/rolldown/tsconfig.check.json
@@ -1,0 +1,20 @@
+{
+  // copied from Vite
+  // https://github.com/vitejs/vite/blob/1465b2064ee23ac5df5414b13355a394ccd931af/packages/vite/tsconfig.check.json
+  "compilerOptions": {
+    "target": "ES2020",
+    "moduleResolution": "node16",
+    "module": "node16",
+    "lib": ["ES2020", "ESNext.Disposable", "DOM"], // ES2020 is very conservative check for client types, could be bumped if needed
+    "types": [], // Avoid checking unrelated node_modules types
+    "noEmit": true,
+    "strict": true,
+    "exactOptionalPropertyTypes": true,
+    "resolveJsonModule": true
+  },
+  "include": [
+    // check only package exports
+    "dist/types/index.d.ts",
+    "dist/types/parallel-plugin.d.ts"
+  ]
+}

--- a/packages/rolldown/tsconfig.check.json
+++ b/packages/rolldown/tsconfig.check.json
@@ -5,16 +5,13 @@
     "target": "ES2020",
     "moduleResolution": "node16",
     "module": "node16",
-    "lib": ["ES2020", "ESNext.Disposable", "DOM"], // ES2020 is very conservative check for client types, could be bumped if needed
-    "types": [], // Avoid checking unrelated node_modules types
+    "lib": ["ES2020", "ESNext.Disposable"], // ES2020 is very conservative check for client types, could be bumped if needed
+    "types": ["node"], // Avoid checking unrelated node_modules types
     "noEmit": true,
     "strict": true,
     "exactOptionalPropertyTypes": true,
     "resolveJsonModule": true
   },
-  "include": [
-    // check only package exports
-    "dist/types/index.d.ts",
-    "dist/types/parallel-plugin.d.ts"
-  ]
+  "include": ["dist/types/**/*.d.ts"],
+  "exclude": []
 }

--- a/packages/rolldown/tsconfig.json
+++ b/packages/rolldown/tsconfig.json
@@ -7,7 +7,7 @@
     "*.cjs",
     "dist",
     "lib",
-    "./src/log/locate-character" // the extracted .d.ts from js files has typing error.
+    "./src/log/locate-character" // the extracted .d.ts from js files has typing error, ignore it because it's not used
   ],
   "compilerOptions": {
     "composite": true,

--- a/packages/rolldown/tsconfig.json
+++ b/packages/rolldown/tsconfig.json
@@ -7,7 +7,7 @@
     "*.cjs",
     "dist",
     "lib",
-    "./src/log/locate-character"
+    "./src/log/locate-character" // the extracted .d.ts from js files has typing error.
   ],
   "compilerOptions": {
     "rootDir": "./src",

--- a/packages/rolldown/tsconfig.json
+++ b/packages/rolldown/tsconfig.json
@@ -10,10 +10,12 @@
     "./src/log/locate-character" // the extracted .d.ts from js files has typing error.
   ],
   "compilerOptions": {
+    "composite": true,
     "rootDir": "./src",
     "outDir": "./dist/types",
     "resolveJsonModule": true,
     "noEmit": false,
+    "tsBuildInfoFile": ".tsbuildinfo",
     "emitDeclarationOnly": true
   }
 }

--- a/packages/rolldown/tsconfig.json
+++ b/packages/rolldown/tsconfig.json
@@ -3,12 +3,12 @@
   "include": ["./src/**/*"],
   "exclude": ["*.js", "*.mjs", "*.cjs", "dist", "lib"],
   "compilerOptions": {
-    "composite": true,
     "rootDir": "./src",
     "outDir": "./dist/types",
     "resolveJsonModule": true,
     "noEmit": false,
-    "tsBuildInfoFile": ".tsbuildinfo",
-    "emitDeclarationOnly": true
+    "emitDeclarationOnly": true,
+    "isolatedModules": true,
+    "isolatedDeclarations": true
   }
 }

--- a/packages/rolldown/tsconfig.json
+++ b/packages/rolldown/tsconfig.json
@@ -1,14 +1,19 @@
 {
   "extends": "../../tsconfig.json",
   "include": ["./src/**/*"],
-  "exclude": ["*.js", "*.mjs", "*.cjs", "dist", "lib"],
+  "exclude": [
+    "*.js",
+    "*.mjs",
+    "*.cjs",
+    "dist",
+    "lib",
+    "./src/log/locate-character"
+  ],
   "compilerOptions": {
     "rootDir": "./src",
     "outDir": "./dist/types",
     "resolveJsonModule": true,
     "noEmit": false,
-    "emitDeclarationOnly": true,
-    "isolatedModules": true,
-    "isolatedDeclarations": true
+    "emitDeclarationOnly": true
   }
 }

